### PR TITLE
Add support for wandb and config file.

### PR DESCRIPTION
**Pull request by dmartinez05**

### CODE REVIEW

**Review by cmarshak — CHANGES_REQUESTED** [review]
The WandDB looks great. Look forward to discussion. Thanks for helping with the environment too. Gattaca2 cluster is definitely the way to go.

I thought we were saving every 5 epochs. Given epochs take a few minutes, it doesn't seem necessary to me to have a Graceful Exit. Can we remove that? At some point, we might want to be able to load weights and rerun epochs, I can see that more useful down the road, but not right now. Goal is to have a very lean training script.

Regarding inspection of validation data - I really don't want that in the training code. Again, want the training code to be easily modifiable and very focused on the training and the NLL loss. I am going to add a notebook with some data inspection but it should really be run at the end. We need to redo how data is curated due to OPERA project requirements and for optimizations. Even now, the data was immediately preceding acquisitions (that's what Harris did) and now we are using anniversary dates.

Is there an obvious way to find the hyperparameters in wandb such that we have maximized NLL at the end of the run? That would be helpful. Might already be there, but just wanted to see.
